scalyr_agent/agent_main.py

```diff
@@ -985,6 +994,61 @@ def __print_force_https_message(self, scalyr_server, raw_scalyr_server):
                 % (raw_scalyr_server, scalyr_server)
             )
 
+    def __get_log_files_initial_positions(self, only_new_files=False):
+        # type: (bool) -> Optional[Dict]
+        """
+        According to the current configuration, determine all agent log files (including main agent.log and
+        multi-process worker sessions logs), create those files (if they do not exist) and get their current positions.
+
+        The preliminarily creation of the log files, in case of their absence, is needed because agent log messages may
+        be written to the agent log file before this log file is processed by the copying manager. In this case,
+        the copying manager will start sending log file from its current position, skipping everything before that and
+        causing data loss.
+
+        :param only_new_files: This option should be used when configuration is reloaded and more worker sessions were
+        added. We ignore existing files and only process log files for new worker sessions.
+        :return:
+        """
+
+        # all paths for all agent log files.
+        log_file_paths_to_check = [self.__log_file_path]
+
+        # we also all log files for all worker session if they are in multiprocess configuration.
+        if self.__config.use_multiprocess_workers:
+            for worker_session_id in self.__config.get_session_ids_from_all_workers():
+                log_file_path = self.__config.get_worker_session_agent_log_path(
+                    worker_session_id
+                )
+                log_file_paths_to_check.append(log_file_path)
+
+        log_file_paths = []
+        for log_file_path in log_file_paths_to_check:
+            include_file = True
+            if os.path.exists(log_file_path):
```

Comment:
Technically and to be more correct, we could also use ``isfile``, but I think it's highly unlikely that directory with such name would exist.